src/c++/perf_analyzer/perf_utils.cc

```diff
@@ -423,4 +424,14 @@ ScheduleDistribution<Distribution::CONSTANT>(const double request_rate)
   return [period](std::mt19937& /*gen*/) { return period; };
 }
 
+TEST_CASE("testing the ParseProtocol function")
```

Comment:
this is the one and only unit test added for proof of concept. future unit tests can be added in future code changes